src/platform_impl/apple/appkit/monitor.rs

```diff
@@ -106,10 +106,110 @@ pub struct MonitorHandle(CGDirectDisplayID);
 
 impl MonitorHandle {
     /// Internal comparisons of [`MonitorHandle`]s are done first requesting a UUID for the handle.
-    fn uuid(&self) -> [u8; 16] {
+    fn uuid(&self) -> u128 {
         let ptr = unsafe { ffi::CGDisplayCreateUUIDFromDisplayID(self.0) };
         let cf_uuid = unsafe { CFRetained::from_raw(NonNull::new(ptr).unwrap()) };
-        unsafe { CFUUIDGetUUIDBytes(&cf_uuid) }.into()
+        u128::from_ne_bytes(unsafe { CFUUIDGetUUIDBytes(&cf_uuid) }.into())
+    }
+
+    pub fn new(id: CGDirectDisplayID) -> Self {
+        MonitorHandle(id)
+    }
+
+    fn refresh_rate_millihertz(&self) -> Option<NonZeroU32> {
+        let current_display_mode =
+            NativeDisplayMode(unsafe { CGDisplayCopyDisplayMode(self.0) }.unwrap());
+        refresh_rate_millihertz(self.0, &current_display_mode)
+    }
+
+    pub fn video_mode_handles(&self) -> impl Iterator<Item = VideoModeHandle> {
+        let refresh_rate_millihertz = self.refresh_rate_millihertz();
+        let monitor = self.clone();
+
+        unsafe {
+            let modes = {
+                let array = CGDisplayCopyAllDisplayModes(self.0, None)
+                    .expect("failed to get list of display modes");
+                let array_count = CFArrayGetCount(&array);
+                let modes: Vec<_> = (0..array_count)
+                    .map(move |i| {
+                        let mode = CFArrayGetValueAtIndex(&array, i) as *mut CGDisplayMode;
+                        CFRetained::from_raw(NonNull::new(mode).unwrap())
```

Comment:
Apply fix from https://github.com/rust-windowing/winit/pull/4151.

```suggestion
                        CFRetained::retain(NonNull::new(mode).unwrap())
```